### DISCUSSION TITLE
How to  achieve an effect similar to SQLCipher

### DISCUSSION THREAD

**Discussion by jueinin** [electric-sql/pglite#243] [Q&A]
access to the database is only granted upon entering the correct password， maybe just like sqlite sqlcipher. we need to encrypt our db file.

**Comment by samwillis** [comment]
Hey @jueinin 

We don't have anything built in for this. A few pointers though:

- we are working on pgcrypto support (encrypt columns) but it's not yet finished: https://github.com/electric-sql/pglite/pull/134
- there is something called Postgres TDE, we it will likely need a custom build of PGLite: https://wiki.postgresql.org/wiki/Transparent_Data_Encryption - We are unlucky to look into this ourselves
- it would be possible to build an encrypted VFS for PGlite, so its encrypted on disk. We are working on docs to show how to build custom VFSs.

**Reply by jueinin** [reply]
@samwillis We are not going to support TDE, right? Does this VFS support IndexedDB?

I want to run pglite in the browser to store some user privacy data, but this data might be exposed. I hope there is a way to protect it.
